### ISSUE TITLE
Query Loop: Omit "No results found" message, if offset is used, and omit entirely if "No results" block is used.

### ISSUE THREAD

**Issue by jasmussen** [WordPress/gutenberg#66548]
When a custom query is applied that's using the "Offset" filter, the use case is commonly a template that has multiple query loops at once, including one that starts _without_ an offset. For example, a "news" type layout:

<div align="center"><img src="https://github.com/user-attachments/assets/a8080444-3b21-42b2-b146-82ee8b24111b" alt="" width="200"></div>

In such a template, the first query loop will already show a "No results found" message. Perhaps it will even show a customized variant using the "No results" block. If you site has 1 or 0 posts, it is not going to be useful to see a repeated "No results" message, for every subsequent off-set loop:

![Image](https://github.com/user-attachments/assets/e9c9c201-ac32-457b-9c51-37d51e184dc6)

Duplicated "No results" blocks were, as shown above, already [removed from the Twenty Twenty-Five News template](https://github.com/WordPress/twentytwentyfive/pull/527). But the "No results found" message comes from the Query Loop block itself, and is output even when a "No results" block is used, being duplicative.

Suggestion:

* [ ] Don't show the "No results found." message if there is a "No results" child block.
* [ ] Don't show the "No results found." message if the query loop has an offset.

**Comment by carolinan** [comment]
My only concern is that without showing anything, it can become difficult to see and understand that there is a query loop block there, so maybe it needs some sort of visual placeholder still, that takes up the space of the block and ensures that it can still be selected with a pointer like a mouse.

**Comment by jasmussen** [comment]
Yes, we could potentially show a placeholder message, similar to what the Content block shows. Such as "No results yet; adjust settings to view." or ideally shorter.

**Comment by JosVelasco** [comment]
While building websites for customers, I find it difficult to explain how the website will look like when they don't have content.

Showing some fake demo posts/placeholders could be excellent with a "demo content" watermark on images and "demo content" titles and excerpts.

**Comment by jasmussen** [comment]
> Showing some fake demo posts/placeholders could be excellent with a "demo content" watermark on images and "demo content" titles and excerpts.

I think that's valid beyond any basic placeholder content for example the "Content" block shows. Careful balance to strike, though, as it's also valid to see how a template looks when your site only has 1 or 2 posts, for example.

You might also be interested in #28466, which suggests a way to "fill out with dummy content" a template, for previewing purposes.
